### PR TITLE
[WIP]Initial Commit

### PR DESCRIPTION
**Pull request by KalyanRally**

### CODE REVIEW

File: implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala [hunk 1]
```diff
@@ -544,12 +544,14 @@ private object MacroImplicits {
           .reduce((a, b) => q"$a || $b")
       }){
                 var i = 0
-                val keys = for{
-                  i <- 0 until ${args.length}
-                  if isMissing(i)
-                } yield i match{
-                  case ..${for (arg <- args)
-        yield cq"${arg.i} => ${arg.mapped}"}
+                val keys = while(i < ${args.length}){
```
Comment (htmldoug) [inline implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala]: I think we need `keys: Seq[String]` which doesn't look like it holds anymore.

Maybe instead we could do something like `(0 until ${args.length}).withFilter(!isMissing(_)).map{ case ..${for (arg <- args) ... } }`. WDYT?

Comment (KalyanRally) [reply implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala]: Made the change now i see different issue
```
[InvariantReturn] This method always returns the same value: 1.
[error]     implicit val pickleProfileResponse = WeePickle.macroFromTo[ProfileResponse]
```

File: implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala [hunk 2]
```diff
@@ -543,13 +543,9 @@ private object MacroImplicits {
           .map(i => q"${foundBitSet(i)} != ${if (i < addlBitSets || args.length > 0 && args.length % 64 == 0) -1 else (1L << args.length) - 1}")
           .reduce((a, b) => q"$a || $b")
       }){
-                var i = 0
-                val keys = for{
-                  i <- 0 until ${args.length}
-                  if isMissing(i)
-                } yield i match{
+                val keys = (0 until ${args.length}).withFilter(!isMissing(_)).map{
```
Comment (htmldoug) [inline implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala]:
```suggestion
                val keys = (0 until ${args.length}).withFilter(isMissing).map{
```

Did `isMissing` got flipped from the original?

File: implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala [hunk 3]
```diff
@@ -543,13 +543,9 @@ private object MacroImplicits {
           .map(i => q"${foundBitSet(i)} != ${if (i < addlBitSets || args.length > 0 && args.length % 64 == 0) -1 else (1L << args.length) - 1}")
           .reduce((a, b) => q"$a || $b")
       }){
-                var i = 0
```
Comment (htmldoug) [inline implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala:546]: Thanks for removing this.

File: implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala (hunk 1 above)
Comment (htmldoug) [reply implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala]: Google points to https://github.com/HairyFotr/linter/blob/7bfe88acad7072c65bffe9c29b51ae72f141bf8b/src/main/scala/Warning.scala#L335-L336.

Any idea which method?

Comment (KalyanRally) [reply implicits/src/com/rallyhealth/weepickle/v1/implicits/MacroImplicits.scala]: From the `sbt compile` below are logs for the code change that was made in this PR

```
val keys = 0.until(37).withFilter(((x$1) => isMissing(x$1).unary_$bang)).map(<empty> match {
                  case 0 => "typeCode"
                  case 1 => "coverageID"
                  case 2 => "designatedNetwork"
                  case 3 => "governmentProgramTypeCode"
                  case 4 => "policyNumber"
                  case 5 => "groupNumber"
                  case 6 => "prodServiceTypeCode"
                  case 7 => "coverageEffDateRange"
                  case 8 => "situsState"
                  case 9 => "fipsCountyCode"
                  case 10 => "fipsLocationCode"
                  case 11 => "lisLevelCode"
                  case 12 => "obligorID"
                  case 13 => "planVariationCode"
                  case 14 => "reportingCode"
                  case 15 => "marketType"
                  case 16 => "marketNumber"
                  case 17 => "fundingArrangement"
                  case 18 => "ancillaryHealthCoverage"
                  case 19 => "productCode"
                  case 20 => "sharedArrangementCode"
                  case 21 => "snpType"
                  case 22 => "ehbIndicator"
                  case 23 => "phiIndicator"
                  case 24 => "pcpEligibilityIndicator"
                  case 25 => "pcpReferralsIndicator"
                  case 26 => "effectuatedIndicator"
                  case 27 => "planCodeValue"
                  case 28 => "doctorPlanCode"
                  case 29 => "privateLabelCustomer"
                  case 30 => "medicalDualStatusCode"
                  case 31 => "hdhpIndicator"
                  case 32 => "productID"
                  case 33 => "exchangeCode"
                  case 34 => "contractNumber"
                  case 35 => "pbpNumber"
                  case 36 => "segmentId"
                });
                throw new com.rallyhealth.weepickle.v1.core.Abort("missing keys in dictionary: ".$plus(keys.mkString(", ")))
```

**Pull request closed without merging**